tests/test_script_helper.py

```diff
@@ -1,222 +1,284 @@
-#    To support more types of synapses, CODE CHANGES are required (lines 157-181).
-# - Global simulation and stimuli parameters (line 31-61) can be changed according to the user's needs.
-
-
-
-from neuron import h, gui
-
-import matplotlib.pyplot as plt
-import math
+'''Test script for subtree reductor:
+ May be used to test the NeuroReduce reduction as performed by
+ subtree_reductor() in subtree_reductor_func.py.  Receives as arguments a
+ Neurolucida full morphology file, a model template file, the total approximate
+ number of segments in the reduced model (-1 for default calculation of a
+ segment for every 0.1 lambda), and a file with synapse locations (see format
+ in Readme) - creates the full morphology instance, reduces it with
+ subtree_reductor(), and compares the voltage traces from the simulation of the
+ original full model and the reduced model using the same random synaptic
+ activity in both simulations, which is determined by parameters in this file.
+
+ How to use: Usage from shell: python test_script.py original_morphology_file
+ model_template_file reduction_frequency manual_total_nsegs
+ synapse_locations_file for example: python test_script.py
+ 2013_03_06_cell08_876_H41_05_Cell2.ASC model.hoc 38 -1
+ origRandomSynapses-10000
+
+ - This tester supports only Neurolucida morphology files. To test NeuroReduce
+   with other kinds of morphology files, CODE CHANGES are required in
+   create_model()
+ - The model template file given to this tester must contain a
+   complete_full_model_creation() function (see Readme and example in the
+                                            attached model.hoc file) to ensure
+   the correct and complete creation of the full model.
+ - This tester supports excitatory and inhibitory synapses - randomly splits
+   the locations in the single given synapse locations file into excitatory and
+   inhibitory synapses according to the distribution determined by the user in
+   line 51 (PERCENTAGE_OF_EXCITATORY_SYNAPSES); gives each synapse's NetStim
+   object a seed that corresponds to the synapse's index as excitatory or
+   inhibitory.  To support more types of synapses, CODE CHANGES are required
+   (lines 157-181).
+ - Global simulation and stimuli parameters (line 31-61) can be changed
+   according to the user's needs.
+'''
+from __future__ import print_function
+
+from collections import namedtuple
+from contextlib import contextmanager
 import sys
 import random
-import numpy as np
-import pdb
 import os
 import subprocess
+from neuron import h
+import numpy as np
+import matplotlib.pyplot as plt
 
-import neuron_reduce
-subtree_reductor =  neuron_reduce.subtree_reductor
-
-APICAL = "apical"
-SOMATIC = "somatic"
+from neuron_reduce import subtree_reductor
 
 #############################
-# simulation parameters - to be adjusted by the user:
-TSTOP = 1500  # ms
-STEPS_PER_MS = 10
-DT = 0.1
+# simulation parameters - to be adjusted by the user
+SIM_PARAMS = {'tstop': 1500,  # ms
```

Comment:
this is part way to not having `globals` for all the settings, in case the tests become more complicated